client/state/themes/actions.js

```diff
@@ -149,19 +156,23 @@ export function receiveThemesQuery( themes, siteId, query ) {
  * @param {number} siteId the Site ID
  * @param {boolean} filterWpcom True to remove all wpcom themes
  * @param {Object} query the theme query
- * @returns {Array} the filtered list of themes
+ * @param {number} found total number of themes matching query
+ * @returns {Object} contains fields filteredThemes and found
  */
-function filterThemes( themes, siteId, filterWpcom, query ) {
+function filterThemes( themes, siteId, filterWpcom, query, found ) {
```

Comment:
Right, that was the problem with the initial PR. I _thought_ we could always calculate the found cound, but of course for non-jetpack sites we need to pass the _total_ found count through to the query manager, otherwise paging (and the displayed found count) will break.